poppy/fresnel.py

```diff
@@ -982,32 +965,52 @@ def _resample_wavefront_pixelscale(self, detector):
             raise NotImplementedError("Resampling to detector doesn't yet work in angular coordinates for Fresnel.")
 
         pixscale_ratio = (self.pixelscale / detector.pixelscale).decompose().value
-        _log.info("Resampling wavefront to detector with {} pixels and {}. Zoom factor is {}".format(
-            detector.shape, detector.pixelscale, pixscale_ratio ))
 
-        _log.debug("Wavefront pixel scale: {}".format(self.pixelscale))
-        _log.debug("Wavefront FOV: {} pixels, {}".format(self.shape, self.shape[0]*u.pixel*self.pixelscale))
+        if np.abs(pixscale_ratio-1.0)< 1e-3:
+            _log.debug("Wavefront is already at desired pixel scale {:.4g}.  No resampling needed.".format(self.pixelscale))
+            return
 
-        _log.debug("Desired detector pixel scale: {}".format(detector.pixelscale))
-        _log.debug("Desired detector FOV: {} pixels, {}".format(detector.shape,
-                                                                      detector.shape[0]*u.pixel*detector.pixelscale,
-                                                                      ))
+        super(FresnelWavefront, self)._resample_wavefront_pixelscale(detector)
 
-        # TODO the following works but is not optimal for performance
-        # We should consider cropping out an appropriate subregion prior to performing the zoom.
-        # That makes a difference if the detector is only sampling a small part of a much larger wavefront
+        self.n = detector.shape[0]
 
-        new_wf_real = scipy.ndimage.zoom(self.wavefront.real, pixscale_ratio, order=detector.interp_order)
-        new_wf_imag = scipy.ndimage.zoom(self.wavefront.imag, pixscale_ratio, order=detector.interp_order)
-        new_wf = new_wf_real + 1.j*new_wf_imag
+    @classmethod
+    def from_wavefront(cls, wavefront):
+        """Convert a Fraunhofer type wavefront object to a Fresnel one
 
-        _log.debug("Cropping/padding resampled wavefront to detector shape")
-        new_wf = utils.pad_or_crop_to_shape(new_wf, detector.shape)
+		Note, this function implicitly assumes this wavefront is at a
```

Comment:
it might be useful to at least have a placeholder for image plane conversion (where the user provides the plate scale).